sci-mathematics/gap/gap-4.12.2-r3.ebuild

```
@@ -0,0 +1,140 @@
+RESTRICT="!test? ( test )"
+
+# The minimum set of packages needed for basic GAP operation. You can
+# actually start gap without these by passing "--bare" to it on the CLI,
+# but don't expect anything to work.
+REQUIRED_PKGS="
+	dev-gap/gapdoc
+	dev-gap/primgrp
+	dev-gap/smallgrp
+	dev-gap/transgrp"
+
+# The packages aren't really required, but GAP tries to load them
+# automatically, and will complain to the user if they fail to load.
+# The list of automatically-loaded packages is a user preference, called
+# AutoloadPackages, and the upstream default can be found in
+# lib/package.gi within the GAP source tree. Passing "-A" to GAP on the
+# CLI (or setting that user preference) will suppress the autoload
+# behavior and allow GAP to start without these, which is why we allow
+# the user to skip them with USE=minimal if he knows what he is doing.
+AUTOLOADED_PKGS="
+	dev-gap/autpgrp
+	dev-gap/alnuth
+	dev-gap/crisp
+	dev-gap/ctbllib
+	dev-gap/factint
+	dev-gap/fga
+	dev-gap/irredsol
+	dev-gap/laguna
+	dev-gap/polenta
+	dev-gap/polycyclic
+	dev-gap/resclasses
+	dev-gap/sophus
+	dev-gap/tomlib"
+
+# The test suite will fail without the "required" subset.
+BDEPEND="test? ( ${REQUIRED_PKGS} )"
+
+DEPEND="dev-libs/gmp:=
+	sys-libs/zlib
+	valgrind? ( dev-debug/valgrind )
+	readline? ( sys-libs/readline:= )"
+
+RDEPEND="${DEPEND}"
+
+# If you _really_ want to install GAP without the set of required
+# packages, use package.provided.
+PDEPEND="${REQUIRED_PKGS} !minimal? ( ${AUTOLOADED_PKGS} )"
+
+pkg_setup() {
+	if use valgrind; then
+		elog "If you enable the use of valgrind during building"
+		elog "be sure that you have enabled the proper flags"
+		elog "in gcc to support it:"
+		elog "https://wiki.gentoo.org/wiki/Debugging#Valgrind"
+	fi
+}
+
+src_prepare() {
+	# Remove these to be extra sure we don't use bundled libraries.
+	rm -r extern || die
+	rm -r hpcgap/extern || die
+
+	# The Makefile just tells you to run ./configure, which then
+	# produces a GNUmakefile.
+	rm Makefile || die
+
+	# Prepend AC_CONFIG_MACRO_DIRS to configure.ac so that eautoreconf
+	# recreates aclocal.m4 correctly. Upstream bundles libtool-2.4.6,
+	# and Gentoo uses 2.4.7, so we need to regenerate aclocal.m4. BUT,
+	# upstream also uses m4_include directives in aclocal.m4 rather than
+	# AC_CONFIG_MACRO_DIRS in configure.ac. Without AC_CONFIG_MACRO_DIRS
+	# eautoreconf will omit all of the macros in cnf/m4.
+	sed -e '1s;^;AC_CONFIG_MACRO_DIRS([cnf/m4])\n;' -i configure.ac || die
+
+	default
+
+	# Fix feature detection with pathological CFLAGS
+	eautoreconf
+}
+
+src_configure() {
+	# We unset $ABI because GAP uses it internally for something else.
+	# --without-gmp and --without-zlib both trigger an AC_MSG_ERROR
+	econf \
+		ABI="" \
+		--with-gmp \
+		--with-zlib \
+		$(use_enable cpu_flags_x86_popcnt popcnt) \
+		$(use_enable memcheck memory-checking) \
+		$(use_enable valgrind) \
+		$(use_with readline) \
+		$(use_enable debug)
```

```suggestion
	local myeconfargs=(
		ABI=""
		--with-gmp
		--with-zlib
		$(use_enable cpu_flags_x86_popcnt popcnt)
		$(use_enable memcheck memory-checking)
		$(use_enable valgrind)
		$(use_with readline)
		$(use_enable debug)
	)
	econf "${myeconfargs[@]}"
```
this could be adapted on other ebuilds too.

---

I did this for sci-mathematics/gap, but none of the other packages need more than a few econf args, making the extra few lines not quite worth the trouble IMO.